### PR TITLE
Support configuration of MongoAuth provider

### LINKED ISSUE #2826: Mongo DB Device Registry: Allow configuration of auth provider properties
Hi,

when using the MongoDB based Device Registry it's possible to enable basic auth using the ``authenticationRequired`` property which will utilize [vertx-auth-mongo](https://vertx.io/docs/vertx-auth-mongo/java/).

This auth provider allows configuration of various [options](https://vertx.io/docs/apidocs/io/vertx/ext/auth/mongo/MongoAuthenticationOptions.html).  (For our use case especially the ``#collectionName``  would be of interest.) 

As far as I can see the configuration is not yet possible, as the provider is initialized with an empty options object:

```org.eclipse.hono.deviceregistry.mongodb.ApplicationConfig#createAuthHandler```:
````
  return BasicAuthHandler.create(
          MongoAuth.create(mongoClient(), new JsonObject()),
          httpServerProperties().getRealm());
````

Would it be possible to make this properties somehow available in the hono registry configuration?

Thanks!
Chris

### PR DESCRIPTION
**Pull request by sophokles73**
The MongAuth provider used to guard access to the Mongo DB based
registry's HTTP endpoint can now be configured using environment
variables.

Fixes #2826

### CODE REVIEW

**Comment by sophokles73** [comment]
@kaniyan would you mind taking a look at this?

**Comment by kaniyan** [comment]
> @kaniyan would you mind taking a look at this?

@sophokles73 I am back on Thursday (16.09) and sure I will take a look. It's also fine for me if someone else would like to take a look before it :)

File: tests/src/test/java/org/eclipse/hono/tests/CrudHttpClient.java [hunk 1]
```diff
@@ -68,11 +68,11 @@ public CrudHttpClient(final Vertx vertx, final String defaultHost, final int def
      * Creates a new client for a host and port.
      *
      * @param vertx The vert.x instance to use.
-     * @param options The client options to use for connecting to servers.
+     * @param httpClientOptions The HTTP client client options to use for connecting to servers.
```
Comment (kaniyan) [inline tests/src/test/java/org/eclipse/hono/tests/CrudHttpClient.java:71]: _client_ is mentioned twice consecutively in the _JavaDoc_.

**Review by kaniyan — CHANGES_REQUESTED** [review]

**Comment by sophokles73** [comment]
@kaniyan I pushed a fix

**Review by kaniyan — APPROVED** [review]
LGTM

**Pull request merged**